employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/common/HasTimeslot.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.domain.common;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+public interface HasTimeslot {
+
+    public static LocalDateTime EPOCH = LocalDateTime.ofEpochSecond(0, 0, ZoneOffset.UTC);
```

Comment:
Noted. Since there is only one shared constant, I don't think it makes sense to create a whole new class for it. For good convention, I'll declare the constant in the separate classes. This should also improve readability for the classes that use `EPOCH`.

Reply:
Declaring the constant in each class that use it leads to code duplication. 

What is the constant used for? Most likely for a comparison or calculation of a duration since the epoch, right? How about encapsulating the constant into a class that has methods providing exactly the described capabilities? The code that now directly use the method can call the methods instead.

Reply:
It is a hack to allow Shift Template to use the same viewer as Shifts and Employee Availabilities. Rotations don't really have/use "Dates", but in order to use the old viewers, a reference to a date was needed to place entities onto the viewer. The reference is the Epoch. Since the frontend and backend code are decoupled now, I think the constant and the `HasTimeslot` interface can be safely removed.

Reply:
Looking through the old code, it seems that with the new frontend and backend, this hack is not needed. I agree with removing it.